lightning/src/ln/peer_handler.rs

```diff
@@ -1088,6 +1094,43 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 			}
 		};
 	}
+
+	/// This function can be called at an interval greater than every 30 seconds, it will:
+	/// disconnect the Peer if awaiting_pong is true
+	/// prepare to Ping all the Peers that the PeerManager is responsible for and set awaiting_pong to true
+	///
+	/// After calling this function one would want to call process_events() in order to finish Pinging the Peer
```

Comment:
You shouldnt need to call process_events() to send - we call send_data on the descriptors in self.do_attempt_write_data.